### PR TITLE
refactor: split pages code

### LINKED ISSUE #183: [Task]: Split pages code into sections
Currently the pages files hold too much code. The page file should only be to manage shared states but the sections should be split into different files to ease maintainance. This way, also if a section can be used somewhere else, we can just import that section in a different page.

### PR DESCRIPTION
**Pull request by cpl121**
Fixes #183 

## Changes

  - Components of nfts and candy machines have been splited from page `candy-machine-ID`
  - Split useMint hook into two - refreshCandyMachine and mint
  
### Checks

* [X] Have you catchup your branch with the latest state in base?
* [X] Have you lint your code locally prior to submission?
* [X] Have you reviewed your proposed changes and removed debris?
* [X] Have you successfully ran due tests with your changes?

### CODE REVIEW

**Review by VmMad — CHANGES_REQUESTED** [review]
It works fine! Great work, just a few comments that I think could improve it a bit and it's done. Btw when you try to update a candy machine but there are no cache files now the message doesn't appear

**Comment by cpl121** [comment]
> It works fine! Great work, just a few comments that I think could improve it a bit and it's done. Btw when you try to update a candy machine but there are no cache files now the message doesn't appear

This error is solved in `feat/improve-pages-error`

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
conflicts with base 🙏🏼

File: components/nft/NftCollection.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            <div className='d-flex flex-justify-start flex-items-center gap-5 mt-3'>
+                <NftCard
+                    title={nft.name}
+                    imageLink={nft.image}
+                    buttons={[
+                        {
+                            text: 'View in Solscan',
+                            as: 'link',
+                            variant: 'invisible',
+                            hash: '14eoYMYLY19gtfE1gwWDhnjDD3fDjGTQTGyicBKT33Ns',
```
Comment (aspnxdd) [inline components/nft/NftCollection.tsx:20]: What is this hash for?

File: hooks/useRefreshCandyMachine.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    async function refreshCandyMachineState() {
+        try {
+            if (!anchorWallet || !wallet.publicKey || !connection || !network) return
+            const provider = new AnchorProvider(connection, anchorWallet, {
+                preflightCommitment: 'recent',
+            })
+
+            const idl = await Program.fetchIdl(CANDY_MACHINE_PROGRAM_V2_ID, provider)
+
+            const program = new Program(idl!, CANDY_MACHINE_PROGRAM_V2_ID, provider)
```
Comment (aspnxdd) [inline hooks/useRefreshCandyMachine.tsx:30]: Instead of adding the ! to the idl, on the line above we could check if there is an idl, else throw an error.

**Review by aspnxdd — CHANGES_REQUESTED** [review]
Much more cleaner now, gj!

File: components/nft/NftCollection.tsx (hunk 1 above)
Comment (cpl121) [reply components/nft/NftCollection.tsx:20]: Link to show example and i forgot to remove it! 😵‍💫

**Review by begonaalvarezd — APPROVED** [review]

**Pull request merged**